src/components/Post/RandomButton.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import axios from 'axios';
+import { useRouter } from 'next/router';
+
+import RandomButton, { postRandomButtonTestIds } from './RandomButton';
+
+jest.mock('axios');
+jest.mock('next/router', () => ({
+  __esModule: true,
+  useRouter: jest.fn(),
+}));
+
+describe('RandomButton', () => {
+  it('renders the button', () => {
+    render(<RandomButton />);
+
+    const randomButton = screen.getByTestId(postRandomButtonTestIds.root);
+
+    expect(randomButton).toBeVisible();
+  });
+
+  describe('when clicked', () => {
+    const mockPush = jest.fn();
+
+    beforeEach(() => {
+      (useRouter as jest.Mock).mockImplementation(() => ({
+        push: mockPush,
+      }));
+    });
+
+    describe('and the returned slug is NOT empty', () => {
+      it('redirects to a random post', async () => {
+        const user = userEvent.setup();
+        const response = {
+          data: {
+            slug: 'example',
+            excludedSlugs: ['example'],
```

Comment:
maybe we could remove this line now?

Reply:
Good catch 👍 
Fixed in https://github.com/nimblehq/til-web/pull/33/commits/0a56f2f3b162d6d38fe44642fba0a52fe5a50856